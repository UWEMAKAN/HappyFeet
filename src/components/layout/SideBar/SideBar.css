.SideBar {
  padding: 0 2rem;
}

@media screen and (max-width: 900px) {
  .SideBar {
    padding: 0;
  }
}

@media screen and (max-width: 450px) {
  .SideBar {
    display: none;
  }
}
