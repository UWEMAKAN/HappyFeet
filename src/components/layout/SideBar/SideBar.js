import React from 'react';
import propTypes from 'prop-types';
import Filters from '../../Filters/Filters';
import './SideBar.css';

const SideBar = ({ selectHandler, searchHandler, checkedHandler }) => (
  <div className="SideBar">
    <Filters
      selectHandler={selectHandler}
      searchHandler={searchHandler}
      checkedHandler={checkedHandler}
    />
  </div>
);

SideBar.propTypes = {
  selectHandler: propTypes.func.isRequired,
  searchHandler: propTypes.func.isRequired,
  checkedHandler: propTypes.func.isRequired
};

export default SideBar;
