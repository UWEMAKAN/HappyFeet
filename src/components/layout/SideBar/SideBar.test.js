import React from 'react';
import { shallow } from 'enzyme';
import SideBar from './SideBar';

const renderSideBar = (args) => {
  const defaultProps = {
    selectHandler: jest.fn(),
    searchHandler: jest.fn(),
    checkedHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<SideBar {...props} />);
};

describe('SideBar Component', () => {
  test('should render SideBar Component', () => {
    const wrapper = renderSideBar();
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
  });
});
