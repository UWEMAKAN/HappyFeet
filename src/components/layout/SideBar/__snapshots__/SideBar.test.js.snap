// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component should render SideBar Component 1`] = `ShallowWrapper {}`;
